docs/source/app_developers_guide/ubuntu.rst

```diff
@@ -2,488 +2,716 @@
+   #. Open a new terminal (the xo terminal window). The prompt ``user@xo-tp$``
+      shows the commands that should be run in this window.
 
-To create and submit the batch containing the new settings,
-enter the following command:
+   #. Run the following command:
 
-.. code-block:: console
+      .. code-block:: console
 
-  $ sawset proposal create sawtooth.validator.transaction_families='[{"family": "intkey", "version": "1.0"}, {"family":"sawtooth_settings", "version":"1.0"}]'
+         user@xo-tp$ sudo -u sawtooth xo-tp-python -v
 
-The output in the validator terminal window includes TP_PROCESS_REQUEST
-messages and information on the authorized keys and transaction families,
-as in this truncated example:
+   #. Check the validator terminal window to confirm that the transaction
+      processor has registered with the validator.
 
-.. code-block:: console
+      .. code-block:: console
 
-  ...
-  [2017-12-05 22:42:46.269 DEBUG    tp_state_handlers] GET: [('000000a87c...\n&sawtooth.settings.vote.authorized_keys\x12B0251fd...
-  ...
-  [2017-12-05 22:42:46.274 DEBUG    tp_state_handlers] GET: [('000000a87c...'sa    wtooth.validator.transaction_families\x12Y[{"family": "intkey", "version": "    1.0"}, {"family":"sawtooth_settings", "version":"1.0"}]')]
-  ...
-  [2017-12-05 22:52:33.495 DEBUG    interconnect] ServerThread sending TP_PROCESS_REQUEST to b'1893abb39b4b4aae'
-  ...
+         [2018-03-14 16:04:18.706 INFO     processor_handlers] registered transaction processor: connection_id=c885e99a11724e04e7da4ee426ee00d4af2cb54b67bf2fbd2f57e862bf28fa2c759a0d0978573782369659124797cc6f38d41bfde2469fe69e7e48dc1fadf5a9, family=xo, version=1.0, namespaces=['5b7349']
 
+   The ``xo-tp-python`` transaction processor continues to run and to display
+   log messages in its terminal window.
 
-Creating and Submitting Transactions
-====================================
+Other Start-up Options for the Transaction Processors
+-----------------------------------------------------
 
-The ``intkey`` command creates sample ``intkey`` (IntegerKey) transactions
-for testing purposes.
+In your own development environment, you could choose other options for starting
+a transaction processor:
 
-This section guides you through the following tasks:
+* You can start the transaction processors before starting the validator.
 
-1. Preparing a batch of ``intkey`` transactions that set the keys
-   to random values.
+* By default, a transaction processor connects to the local validator on port
+  4004. You can change this by specifying a different endpoint as an argument.
+  The following command is equivalent to the default value:
 
-2. Generating *inc* (increment) and *dec* (decrement) transactions to apply to
-   the existing state stored in the blockchain.
+  .. code-block:: console
 
-3. Submitting these transactions to the validator.
+     user@intkey-tp$ sudo -u sawtooth intkey-tp-python -v tcp://127.0.0.1:4004
```

Comment:
This example is incorrect. It should be `--connect tcp://127.0.0.1:4004`. Also, the exact CLI arguments for transaction processors is implementation dependent and they are only consistent by convention, which should probably be mentioned. A link to the CLI reference would also be appropriate here.